src/main/java/ru/yandex/practicum/filmorate/service/DirectorService.java

```diff
@@ -0,0 +1,47 @@
+package ru.yandex.practicum.filmorate.service;
+
+import lombok.NonNull;
+import lombok.RequiredArgsConstructor;
+import org.springframework.beans.factory.annotation.Qualifier;
+import org.springframework.stereotype.Service;
+import ru.yandex.practicum.filmorate.dao.DirectorStorage;
+import ru.yandex.practicum.filmorate.exception.DirectorNotFoundException;
+import ru.yandex.practicum.filmorate.model.Director;
+
+import java.util.List;
+
+@Service
+@RequiredArgsConstructor
+public class DirectorService {
+    @Qualifier("directorDbStorage")
+    @NonNull
+    private final DirectorStorage directorStorage;
+
+    public List<Director> get() {
+        return directorStorage.get();
+    }
+
+    public Director getById(int id) {
+        checkIfDirectorExists(id);
+        return directorStorage.getById(id).get();
+    }
+
+    public Director create(Director director) {
+        return directorStorage.create(director);
+    }
+
+    public Director update(Director director) {
+        checkIfDirectorExists(director.getId());
+        return directorStorage.udpate(director).get();
+    }
+
+    public void delete(int id) {
+        directorStorage.delete(id);
+    }
+
+    public void checkIfDirectorExists(int id) {
+        directorStorage.getById(id)
```

Comment:
Соответственно тут нужно просто добавить `return`:
```suggestion
        return directorStorage.getById(id)
```
Так как `orElseThrow` разворачивает `Optional`, если он не пуст, или выбрасывает исключение, когда `Optional` пустой

Соответственно метод `checkIfDirectorExists` нужно будет переименовать в `getIfDirectorExists`, и сразу возвращать результат работы данного метода в `getById` и `update`

Reply:
Исправлено: метод `checkIfDirectorExists` переименован в `getIfDirectorExists`, метод `getById` в `DirectorService` оптимизирован. 
По поводу метода `update`: возвращать результат работы метода `getIfDirectorExists` там нельзя. Логика метода `update` такова:
1. С помощью метода `getIfDirectorExists` проверяется, существует ли в БД режиссер, параметры которого нужно изменить (при этом результатом работы метода `getIfDirectorExists` будет текущий, еще не измененный в БД режиссер).
2. Если режиссер существует, то на уровне DAO вносятся изменения найденному режиссеру (в противном случае, выбрасывается ошибка).
3. После внесения изменений метод `update` возвращает измененного режиссера.